### DISCUSSION TITLE
new `[editor.file-picker]` key for ignoring a list of specific directories and files that are not always part of a repository tracked by a VCS

### DISCUSSION THREAD

**Discussion by bindsdev** [helix-editor/helix#5494] [Ideas] [answered]
# Issue

I have a miscellaneous Rust (Cargo) package where I write random code to experiment with things for fun. I did not initialize this Cargo package with a VCS since it will not be pushed anywhere and there will be no meaningful code to push. But, when using the file picker, I still am able to pick from anything in the `target/` directory of this Rust project. It is quite annoying to see all of them as options there.

# Solution

My proposed solution for this is to add a new key for the `[editor.file-picker]` configuration section that allows the user to specify a list of directories and files to always prevent from being show in the file picker, in the case that there might not be a version control system initialized in specific contexts. This key could have a name along the lines of `unwated-sometimes-no-vcs` (pretty bad name, but it could always be improved) to show that the key specifies a list of directories or files that might not always be present in a version control environment.

*P.S.*: I am relatively new to the Helix editor, so if there is any existing solution to this I would love to know! Nothing described in the configuration documentation for the [`[editor.file-picker]` section](https://docs.helix-editor.com/configuration.html#editorfile-picker-section) seemed like it would help this situation.

**Comment by lunfel** [comment, marked as answer]
I have stumbled on some similar issue just now. You can check at the configurations here https://docs.helix-editor.com/configuration.html#editorfile-picker-section.

You can add .ignore file at the root of your workspace (works similarly to .gitignore) and will prevent files and folders to show up in the file picker.

My understanding is that .ignore is some sort of alternative to .gitignore when you are not in a git tracked repository. So in your case .ignore file should work fine.

**Reply by bindsdev** [reply]
I will definitely see if that works out. Thank you very much.

**Comment by pergamomo** [comment]
I work at a company where we can't just add editor specific files to all our repositories. Is it possible to add this to the helix config?

**Reply by paholg** [reply]
You can put ignore files in various locations, including in your home directory and Helix config directory. See the text below this table: https://docs.helix-editor.com/editor.html#editorfile-picker-section
